xarray/tests/test_indexing.py

```diff
@@ -63,30 +64,59 @@ def test_group_indexers_by_index(self) -> None:
         data.coords["y2"] = ("y", [2.0, 3.0])
 
         indexes, grouped_indexers = indexing.group_indexers_by_index(
-            data, {"z": 0, "one": "a", "two": 1, "y": 0}
+            data, {"z": 0, "one": "a", "two": 1, "y": 0}, {}
         )
-        assert indexes == {"x": data.xindexes["x"], "y": data.xindexes["y"]}
-        assert grouped_indexers == {
-            "x": {"one": "a", "two": 1},
-            "y": {"y": 0},
-            None: {"z": 0},
-        }
+        for k in indexes:
+            if indexes[k].equals(data.xindexes["x"]):
+                assert grouped_indexers[k] == {"one": "a", "two": 1}
+            elif indexes[k].equals(data.xindexes["y"]):
+                assert grouped_indexers[k] == {"y": 0}
+        assert grouped_indexers[None] == {"z": 0}
+        grouped_indexers.pop(None)
+        assert indexes.keys() == grouped_indexers.keys()
 
         with pytest.raises(KeyError, match=r"no index found for coordinate y2"):
-            indexing.group_indexers_by_index(data, {"y2": 2.0})
+            indexing.group_indexers_by_index(data, {"y2": 2.0}, {})
         with pytest.raises(KeyError, match=r"w is not a valid dimension or coordinate"):
-            indexing.group_indexers_by_index(data, {"w": "a"})
+            indexing.group_indexers_by_index(data, {"w": "a"}, {})
         with pytest.raises(ValueError, match=r"cannot supply.*"):
-            indexing.group_indexers_by_index(data, {"z": 1}, method="nearest")
+            indexing.group_indexers_by_index(data, {"z": 1}, {"method": "nearest"})
 
-    def test_remap_label_indexers(self) -> None:
-        def test_indexer(data, x, expected_pos, expected_idx=None) -> None:
-            pos, new_idx_vars = indexing.remap_label_indexers(data, {"x": x})
-            idx, _ = new_idx_vars.get("x", (None, None))
-            if idx is not None:
-                idx = idx.to_pandas_index()
-            assert_array_equal(pos.get("x"), expected_pos)
-            assert_array_equal(idx, expected_idx)
+    def test_remap_label_indexers(self):
+        def test_indexer(
+            data,
+            x,
+            expected_pos,
+            expected_idx=None,
+            expected_vars=None,
+            expected_drop=None,
+            expected_rename_dims=None,
+        ):
```

Comment:
```suggestion
    def test_remap_label_indexers(self) -> None:
        def test_indexer(
            data,
            x,
            expected_pos,
            expected_idx=None,
            expected_vars=None,
            expected_drop=None,
            expected_rename_dims=None,
        ) -> None:
```
The typing got lost in the merge?

Reply:
No it is a temporary dirty fix until I can figure out why I get a `too many arguments` mypy error. I checked the calls to `test_indexer` below and didn't see anything wrong. Maybe it has to do with unpacked arguments, something similar to https://github.com/python/mypy/issues/9007?

Reply:
I think you can do: 

```python
        a, b = PandasIndex.from_pandas_index(pd.Index([-1, -2]), "three")
        test_indexer(
            mdata,
            ("a", 1),
            [True, True, False, False, False, False, False, False],
            a,
            b,
            ["x", "one", "two"],
            {"x": "three"},
        )
```
Unnecessary workaround but that's how it is with mypy. 

Sometimes I wonder if it's better to add another type checker and IF both of them fail, then the CI is allowed to fail.

Reply:
The better solution is to type the output of `from_pandas_index`, for example like this:
```python
    @classmethod
    def from_pandas_index(cls, index: pd.Index, dim: Hashable) -> Tuple[Any, Any]:
```
I didn't figure what the output was so I used Any instead, but you have a better feel for that.